packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js

```diff
@@ -1121,6 +1121,26 @@ QUnit.module('keyboard navigation', {
         assert.equal(tabsFocusedIndex, $(this.instance.option('focusedElement')).index(), 'multiView focused element is equal tabs focused element');
     });
 
+    QUnit.test('should go back to first index when loop = true and root element is focused', function(assert) {
```

Comment:
Why do you add "root element is focused"? As far as I see focus is not important here, but important that we change loop at runtime and swipe before kbn
```suggestion
    QUnit.test('looping should work on keyboard navigation after loop runtime change to true and swipe', function(assert) {
```

Reply:
[f38015f](https://github.com/DevExpress/DevExtreme/pull/28303/commits/f38015f174790a18cc848f8168582fdbd0f0db6d)